### DISCUSSION TITLE
Multi-chain configuraton for WalletConnect

### DISCUSSION THREAD

**Discussion by devrus91** [WalletConnect/walletconnect-monorepo#3706] [Developer Support] [answered]
I have some problem with wagmi and WalletConnect
Here is my initial configuration:
```
import { EthereumClient, w3mConnectors, w3mProvider } from '@web3modal/ethereum'
import { configureChains, createConfig, WagmiConfig } from 'wagmi'
import { arbitrum, mainnet, polygon } from 'wagmi/chains'

const supportedChains = [mainnet, polygon]

const { publicClient, chains, webSocketPublicClient } = configureChains(supportedChains, [w3mProvider({ projectId })])
const connectors = w3mConnectors({projectId, chains})

const wagmiConfig = createConfig({
  autoConnect: true,
  connectors,
  publicClient, 
  webSocketPublicClient
})

export const ethereumClient = new EthereumClient(wagmiConfig, chains)
```

I am using Trust Wallet. I have imported in it my metamask wallet via private key as Etherium and Polygon separate wallets. So my wallets are not multi-chain wallets.

When trying to connect with Poligon wallet I receive error: ```Requested chains are not supported```
If I change the order of supportedChains like this: ```const supportedChains = [polygon, mainnet]``` than Polygon wallet is accespted while Etherium was rejected with same error ```Requested chains are not supported```

Is this a bug or do i have some problems in configuration?

**Comment by glitch-txs** [comment, marked as answer]
the way wagmi works is, first chain of the array gets added as required chain of the session, that's by it needs to be supported, we will remove this config and move everything to optional chains soon.
For now you can pass the chain you want to connect to into the wagmi's connect function and that chain is going to override the required one.

**Comment by divine-comedian** [comment]
Hey @glitch-txs - Can you explain this a bit more? 

I'm experiencing a similar issue trying to use web3modal - throughout our dapp we connect users with `useWeb3Modal` hook. We do not use the connect functions provided by wagmi and instead rely on web3modal. 

Are you saying we should use the connect function from the useConnect hook of wagmi and specify only the compatible chain?

**Reply by glitch-txs** [reply]
Hi @divine-comedian for Web3Modal SDK you can use the network view so users can choose the required chain. Are you using v2 or v3?

**Reply by divine-comedian** [reply]
Oh, interesting!

V3.0.2

**Reply by glitch-txs** [reply]
You can use web components https://docs.walletconnect.com/web3modal/react/components#w3m-account-button-
Or the hook https://docs.walletconnect.com/web3modal/react/hooks#useweb3modal

If you're using another framework you can find similar functions there :)

**Reply by divine-comedian** [reply]
Ah cool! I'll take a look 

And perhaps sort of related, is there a way to limit which networks a user could connect to for a given wallet?

**Reply by glitch-txs** [reply]
for WalletConnect the session proposal will only add the chains you added in the initial configuration. For extension wallets you can track the network with Wagmi hooks
